openverse_catalog/dags/providers/provider_workflow_types.py

```diff
@@ -0,0 +1,167 @@
+    start_date:        datetime.datetime giving the first valid execution
+                       date of the DAG.
+    max_active_tasks:  integer that sets the number of tasks which can
+                       run simultaneously for this DAG, and the number of
+                       dagruns of this DAG which can be run in parallel.
+                       It's important to keep the rate limits of the
+                       Provider API in mind when setting this parameter.
+    schedule_string:   string giving the schedule on which the DAG should
+                       be run.  Passed to the airflow.dag.DAG __init__
+                       method.
+    dated:             boolean giving whether the `main_function` takes a
+                       string parameter giving a date (i.e., the date for
+                       which data should be ingested).
+    day_shift:         integer giving the number of days before the
+                       current execution date the `main_function` should
+                       be run (if `dated=True`).
+    execution_timeout: datetime.timedelta giving the amount of time a given data
+                       pull may take.
+    doc_md:            string which should be used for the DAG's documentation markdown
+    media_types:       list describing the media type(s) that this provider handles
+                       (e.g. `["audio"]`, `["image", "audio"]`, etc.)
+    """
+
+    dag_id: str
+    provider_script: str
+    default_args: Optional[Dict] = None
+    start_date: datetime = datetime(1970, 1, 1)
+    max_active_tasks: int = 1
+    schedule_string: str = "@monthly"
+    dated: bool = False
+    day_shift: int = 0
+    execution_timeout: timedelta = timedelta(hours=24)
+    doc_md: str = ""
+    media_types: Sequence[str] = ("image",)
+
+
+PROVIDER_WORKFLOWS = [
+    ProviderWorkflow(
+        dag_id="brooklyn_museum_workflow",
+        provider_script="brooklyn_museum",
+        start_date=datetime(2020, 1, 1),
+    ),
+    ProviderWorkflow(
+        dag_id="cleveland_museum_workflow",
+        provider_script="cleveland_museum_of_art",
+        start_date=datetime(2020, 1, 15),
+        execution_timeout=timedelta(hours=12),
+    ),
+    ProviderWorkflow(
+        dag_id="europeana_workflow",
+        provider_script="europeana",
+        schedule_string="@daily",
+        dated=True,
+    ),
+    ProviderWorkflow(
+        dag_id="finnish_museums_workflow",
+        provider_script="finnish_museums",
+        start_date=datetime(2020, 9, 1),
+        execution_timeout=timedelta(days=3),
+    ),
+    ProviderWorkflow(
+        dag_id="flickr_wokrflow",
+        provider_script="flickr",
+        schedule_string="@daily",
+        dated=True,
+    ),
+    ProviderWorkflow(
+        dag_id="freesound_workflow",
+        provider_script="freesound",
+        # doc_md = freesound.__doc__,
+        media_types=("audio",),
+    ),
+    ProviderWorkflow(
+        dag_id="jamendo_workflow",
```

Comment:
We could actually add `__post_init__` logic to the `NamedTuple` which would compute the `dag_id`! It could also be overridden for cases where it's not `f"{provider_script}_workflow"`.

Reply:
That's a good idea! We'd just have to rename a few of the provider scripts to avoid losing DAG history. I think rawpixel, Metropolitan, and Walters. But I see no reason not to!